### PR TITLE
Carolina

### PR DESCRIPTION
**Pull request by carolarruda**

### CODE REVIEW

File: src/components/styles/style.css [hunk 1]
```diff
@@ -192,5 +201,5 @@ input[type="number"] {
 .error {
   font-size: 13px;
   color: red;
-align-self: center;
-}
+  align-self: center;
```
Comment (procrastian) [inline src/components/styles/style.css:204]: Nice that you were able to incorporate some CSS into this!

File: src/server/controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+const logUser = async (req, res) => {
+  const { username, password } = req.body;
+  const user = await prisma.user.findUnique({
+    where: { username: username },
+    select: { password: true },
+  });
+  const createToken = (payload, secret) => {
+    const token = jwt.sign(payload, secret);
+    return token;
+  };
```
Comment (procrastian) [inline src/server/controllers/user.js:35]: You could also use this to create a token on register - that way when a user registers they don't have to log in as well

File: src/server/controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  function hasAccess(result) {
```
Comment (procrastian) [inline src/server/controllers/user.js:37]: Nice way of separating out authentication as part of the login function

File: src/server/controllers/movie.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+const { Prisma } = require("@prisma/client");
+const prisma = require("../utils/prisma");
+
+const getMovies = async (req, res) => {
+  const getMovies = await prisma.movie.findMany();
+  return res.send({ movies: getMovies });
+};
+const createMovie = async (req, res) => {
```
Comment (procrastian) [inline src/server/controllers/movie.js:8]: Is the server checking that the user is logged in before it allows a movie to be created? Or is it all handled as part of only rendering the movie options if there is a token? That's a smart way around it!

**Review by procrastian — COMMENTED** [review]
Awesome work Carolina! You've gone above and beyond on this one! Really like how you've separated out your files into single responsibility, and maintained clean and readable code!

File: src/server/controllers/user.js (hunk 2 above)
Comment (carolarruda) [reply src/server/controllers/user.js:35]: They don't have to, my fetch is taking care of that. If someone register they login as well when registering.

File: src/server/controllers/movie.js (hunk 4 above)
Comment (carolarruda) [reply src/server/controllers/movie.js:8]: The latter, I also clear the localStorage if you go back to login to give a sense of security, same with my Logout button

File: src/server/controllers/user.js (hunk 2 above)
Comment (procrastian) [reply src/server/controllers/user.js:35]: That's smart! I couldn't see that in your register function, could you explain to me how you handle it?

File: src/server/controllers/movie.js (hunk 4 above)
Comment (procrastian) [reply src/server/controllers/movie.js:8]: Nice! Good to see different approaches, I've gone with verifying on server side as well as I don't trust users :P

File: src/server/controllers/user.js (hunk 2 above)
Comment (carolarruda) [reply src/server/controllers/user.js:35]: `    fetch("http://localhost:4000/user/register", opts)
      .then((res) => res.json())
      .then(() => {
        setUsers(users);
        fetch("http://localhost:4000/user/register")
          .then((res) => res.json())
          .then((data) => setUsers(data));
        fetch("http://localhost:4000/user/login", opts)
          .then((res) => res.json())
          .then((data) => {
            setLoggedIn(true);
            localStorage.setItem("token", data.data.token);
            localStorage.setItem("username", data.data.username);
            fetch("http://localhost:4000/movie")
              .then((res) => res.json())
              .then((data) => {
                navigate(`/main`);
                setMovies(data.movies);
              });
          });
      });`
      My third fetch is for the login and I use the same opts for both register and login. It doesn't look pretty but it works.

Comment (carolarruda) [reply src/server/controllers/user.js:35]: My third fetch is for the login and I use the same opts for both register and login. It doesn't look pretty but it works.

File: prisma/schema.prisma [hunk 5]
```diff
@@ -12,18 +12,19 @@ datasource db {
...
-  title         String      @unique
-  description   String
-  runtimeMins   Int
-  createdAt     DateTime    @default(now())
-  updatedAt     DateTime    @updatedAt
+  id          Int      @id @default(autoincrement())
+  title       String   @unique
+  description String
+  runtimeMins Int
+  url         String
```
Comment (marc-alzapiedi) [inline prisma/schema.prisma:27]: Have you added an extra feature to the form to add a url? very nice

File: src/client/App.jsx [hunk 6]
```diff
@@ -1,13 +1,55 @@
...
-    </div>
+  useEffect(() => {
+    if (token) {
+      setLoggedIn(true);
+      fetch("http://localhost:4000/movie")
+        .then((res) => res.json())
+        .then((data) => {
+          setMovies(data.movies);
+        })
+        .catch((error) => console.error("Error fetching movies:", error));
```
Comment (marc-alzapiedi) [inline src/client/App.jsx:26]: Ok would have thought of doing the error handling in the app, cool

File: src/client/App.jsx [hunk 7]
```diff
@@ -1,13 +1,55 @@
...
+              loggedIn && (
+                <>
+                  <MovieList movies={movies} setMovies={setMovies} />
+                </>
+              )
+            }
+          />
+          <Route
+            path="/user/register"
+            element={<RegisterUI setMovies={setMovies} />}
```
Comment (marc-alzapiedi) [inline src/client/App.jsx:47]: Oh wow so you've upgraded the front end react app with routes, ok advanced

File: src/components/MovieList.js [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+      setRuntimeMins(e.target.value);
+    }
+  };
+
+  const handleUrl = (e) => {
+    setUrl(e.target.value);
+  };
+  const handleLogout = () => {
+    navigate(`/`);
+    setLoggedIn(false);
```
Comment (marc-alzapiedi) [inline src/components/MovieList.js:29]: ok so you've even implemented logout, it seems you've made a fully functional webpage

File: src/components/LoginUI.js [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+            <button className="sign" onClick={handleRegister}>
+              Sign up
+            </button>
+          </div>
+
+          <form onSubmit={handleSubmit} className="formy">
+            <label htmlFor="username">Username</label>
+            <input
+              type="text"
+              id="username"
```
[the remark this answers is not in the document]
Comment (carolarruda) [reply src/components/LoginUI.js:117]: Gonna definitely do this when refactoring.

**Pull request closed without merging**